vllm/model_executor/layers/fused_moe/modular_kernel.py

```diff
@@ -834,11 +864,18 @@ def forward(
                 self.fused_experts.quant_config,
             )
 
-            assert self.shared_experts is not None
-            shared_output = self.shared_experts(a1)
+            # assert self.shared_experts is not None
+            if self.shared_experts is not None:
+                shared_output = self.shared_experts(a1)
+
+            dbo_register_recv_hook(hook)
+            dbo_yield()
+
+            if dbo_enabled():
+                hook = None
```

Comment:
Why?

Reply:
I understand the codeflow now by why do you not want to pass the `hook` to receiver? Maybe a comment would be helpful